tests/strategies/test_ddp_strategy.py

```diff
@@ -127,3 +127,24 @@ def test_ddp_configure_ddp():
     trainer.strategy.setup(trainer)
     # in DDPStrategy configure_ddp(), model are still LightningModule
     assert isinstance(trainer.model, LightningModule)
+
+
+class CheckOptimizerDeviceModel(BoringModel):
+    def configure_optimizers(self):
+        assert all(param.device.type == "cuda" for param in self.parameters())
+        super().configure_optimizers()
+
+
+@RunIf(min_gpus=1)
+@pytest.mark.parametrize("strategy", ("ddp", "ddp_spawn"))
```

Comment:
I added both tests cases into this file to avoid duplication, and once we merge the DDPStrategy with DDPSpawnStrategy (after #11643), this anyway collapses to this same test.